repo/mem_repo.go

```diff
@@ -35,13 +34,23 @@ type MemRepo struct {
 
 var _ Repo = (*MemRepo)(nil)
 
+// NewMemRepoProfile creates a new in-memory repository and an empty profile
+// store owned by the given profile
+func NewMemRepoProfile(ctx context.Context, owner *profile.Profile, fs *muxfs.Mux, bus event.Bus) (*MemRepo, error) {
```

Comment:
The way this names reads makes it seems like it's returning a "repo profile" (a profile where repo is an adjective describing it), when its really more of a "repo and profile", or "repo with profile". Would prefer something like `NewMemRepoWithProfile` or perhaps `NewOwnedMemRepo` or something like that.